File: src/dotnet/Orders.Domain/Models/OrderLine.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Orders.Domain.Models;

public class OrderLine : BaseEntity
{
    public long Price { get; set; }
    public int Quantity { get; set; }
    [Range(0, 100)]
    public int Discount { get; set; }

    [ForeignKey(nameof(Order))]
    public Guid OrderId { get; set; }
    public Order? Order { get; set; }

    [ForeignKey(nameof(Product))]
    public Guid ProductId { get; set; }
    public Product? Product { get; set; }
}

File: src/client/src/models/orderLine.ts
import { Guid } from "guid-typescript";
import { randomInt } from "../util";
import { Order } from "./order";
import { Product } from "./product";

export interface OrderLine {
    id: string;
    price: number;
    quantity: number;
    discount: number;
    orderId?: string;
    order?: Order | null;
    productId?: string;
    product?: Product | null;
}

export class OrderLineFactory {
    static create(amount: number, products: Product[]) {
        const orderLines: OrderLine[] = [];

        for (let i = 0; i < amount; i++) {
            const product = products[randomInt(0, products.length - 1)];

            const orderLine: OrderLine = {
                id: Guid.create().toString(),
                price: product.price,
                quantity: randomInt(1, 5),
                discount: randomInt(0, 100),
                productId: product.id,
            }

            orderLines.push(orderLine);
        }

        return orderLines;
    }
}